File: lib/domain/api_keys/gateways/user_api_key_rotation_gateway.rb
# frozen_string_literal: true

module Domain
  module ApiKeys
    module Gateways
      class UserApiKeyRotationGateway
        # @return [Hash] keys: :ok (Boolean), :api_key (String, nil), :error (nil | :not_found)
        def rotate(user_id:, regenerate:)
          raise NotImplementedError, "Subclasses must implement rotate"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/api_keys/gateways/user_api_key_rotation_gateway.rs
use crate::api_keys::dtos::UserApiKeyRotationOutput;

/// Ruby: `Domain::ApiKeys::Gateways::UserApiKeyRotationGateway`
pub trait UserApiKeyRotationGateway: Send + Sync {
    fn rotate(&self, user_id: i64, regenerate: bool) -> UserApiKeyRotationOutput;
}
